## 06-fix-im-command.patch by Christian Welzel <gawain@camlann.de>
##
## DP: fix the command used to quote the imagemagick command for shell invocation.

--- typo3.orig/t3lib/utility/class.t3lib_utility_command.php
+++ typo3/t3lib/utility/class.t3lib_utility_command.php
@@ -86,7 +86,7 @@
 				setlocale(LC_CTYPE, $GLOBALS['TYPO3_CONF_VARS']['SYS']['systemLocale']);
 			}
 			$originalPath = $path . 'gm' . $isExt;
-			$path = escapeshellarg($originalPath);
+			$path = escapeshellcmd($originalPath);
 			if ($GLOBALS['TYPO3_CONF_VARS']['SYS']['UTF8filesystem']) {
 				setlocale(LC_CTYPE, $currentLocale);
 			}
@@ -105,7 +105,7 @@
 				setlocale(LC_CTYPE, $GLOBALS['TYPO3_CONF_VARS']['SYS']['systemLocale']);
 			}
 			$originalPath = $path . (($command == 'composite') ? $combineScript : $command) . $isExt;
-			$path = escapeshellarg($originalPath);
+			$path = escapeshellcmd($originalPath);
 			if ($GLOBALS['TYPO3_CONF_VARS']['SYS']['UTF8filesystem']) {
 				setlocale(LC_CTYPE, $currentLocale);
 			}
@@ -147,4 +147,4 @@
 
 }
 
-?>
\ No newline at end of file
+?>
